j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/Javac.java

```diff
@@ -33,28 +40,70 @@
     StandardJavaFileManager fileManager;
     private DiagnosticCollector<JavaFileObject> listener;
 
-    public Javac(BuildLog log, File generatedClassesPath, List<File> sourcePaths, List<File> classpath, File classesDirFile, File bootstrap) throws IOException {
+    public Javac(BuildLog log, File generatedClassesPath, List<File> sourcePaths, List<File> classpath, File classesDirFile, File bootstrap, Mode mode) throws IOException {
         this.log = log;
-//        for (File file : classpath) {
-//            System.out.println(file.getAbsolutePath() + " " + file.exists() + " " + file.isDirectory());
-//        }
-        javacOptions = new ArrayList<>(Arrays.asList("-encoding", "utf8", "-implicit:none", "-bootclasspath", bootstrap.toString()));
+        compiler = ToolProvider.getSystemJavaCompiler();
+        listener = new DiagnosticCollector<>();
+        fileManager = compiler.getStandardFileManager(listener, null, StandardCharsets.UTF_8);
+        javacOptions = new ArrayList<>(Arrays.asList("-encoding", "utf8", "-implicit:none"));
+        if (log.isDebugEnabled()) {
+            javacOptions.add("-verbose");
+        }
         if (generatedClassesPath == null) {
             javacOptions.add("-proc:none");
         }
+
+        final String javaVersion = "java.specification.version:" + System.getProperty("java.specification.version");
+        String bootClassPath = bootstrap.getCanonicalFile().getAbsolutePath();
+
         if (SourceVersion.latestSupported().compareTo(SourceVersion.RELEASE_8) > 0) {
-            //java 9+
-            javacOptions.add("--release=8");
+            // Java version >= 9
+            //javacOptions.add("--release=8");
```

Comment:
if the line isn't used, please remove it

Reply:
Why was it there? release=8 is incompatible with --add-reads, therefore I had to remove it. But maybe setting release=8 had some other intention? If I understood j2cl compiler correctly, it reads the sources of the current project and the classes (type descriptions) from Java dependencies. So what is the highest Java class format it can read? What trouble could be caused by removing release=8?